ClassFieldTheory/Mathlib/Algebra/Module/Torsion/Basic.lean

```diff
@@ -0,0 +1,32 @@
+import Mathlib.Algebra.Module.Torsion.Basic
+
+@[simps]
+def Module.IsTorsionBy.coprime_decompose {k m1 m2: ℕ} {R M : Type*} [CommRing R] [AddCommGroup M]
```

Comment:
I am not clear about whether this is ever useful. 

I think the interesting question is how to get from "`H^2(G,L^*)` is finite and has size at most |G| for G a p-group" to "`H^2(G,L^*)` is finite and has size at most |G| for G any group". The proof on paper is "`H^2(G,L^*)` is killed by |G| from an earlier result, so injects into prod_{p divides |G|} `H^2(G,L^*)`[p^infty] which injects into prod_{p divides |G|} `H^2(P,L^*)` where P is a Sylow p-subgroup and now done". But I am still quite unclear about how this will be done in Lean.